harness/tests/integration_cases/test_raft.rs

```diff
@@ -212,6 +211,166 @@ fn test_progress_update() {
+    nt.send(vec![new_message(2, 2, MessageType::MsgPropose, 2)]);
+    nt.recover();
+    nt.dispatch(vec![new_message(2, 2, MessageType::MsgPropose, 1)])
+        .expect("");
+
+    // [msg_type: MsgAppend to: 1 from: 2 term: 2 log_term: 2 index: 6 entries {term: 2 index: 7 data: "somedata"} commit: 4,
+    // msg_type: MsgAppend to: 3 from: 2 term: 2 log_term: 2 index: 6 entries {term: 2 index: 7 data: "somedata"} commit: 4]
+    let msg_append = nt.read_messages();
+
+    nt.dispatch(msg_append).expect("");
+
+    // [msg_type: MsgAppendResponse to: 2 from: 1 term: 2 index: 6 commit: 4 reject: true reject_hint: 4,
+    // msg_type: MsgAppendResponse to: 2 from: 3 term: 2 index: 6 commit: 4 reject: true reject_hint: 4]
+    let msg_append_response = nt.read_messages();
+
+    nt.dispatch(msg_append_response).expect("");
+
+    // [msg_type: MsgAppend to: 3 from: 2 term: 2 log_term: 2 index: 4 entries {term: 2 index: 5 data: "somedata"} entries {term: 2 index: 6 data: "somedata"} entries {term: 2 index: 7 data: "somedata"} commit: 4,
+    // msg_type: MsgAppend to: 1 from: 2 term: 2 log_term: 2 index: 4 entries {term: 2 index: 5 data: "somedata"} entries {term: 2 index: 6 data: "somedata"} entries {term: 2 index: 7 data: "somedata"} commit: 4]
+    let msg_append = nt.read_messages();
+
+    // committed index remain the same
+    assert_eq!(
+        (
+            nt.peers[&2].prs().get(1).unwrap().committed_index,
+            nt.peers[&2].prs().get(2).unwrap().committed_index,
+            nt.peers[&2].prs().get(3).unwrap().committed_index
+        ),
+        (4, 0, 4)
+    );
+
+    // resend append
+    nt.send(msg_append);
+
+    // log is up-to-date
+    assert_eq!(
+        (
+            nt.peers[&2].prs().get(1).unwrap().committed_index,
+            nt.peers[&2].prs().get(2).unwrap().committed_index,
+            nt.peers[&2].prs().get(3).unwrap().committed_index
+        ),
+        (7, 0, 7)
+    );
+
+    // set node 1 as Leader again
+    nt.send(vec![new_message(1, 1, MessageType::MsgHup, 0)]);
+    assert_eq!(nt.peers[&1].state, StateRole::Leader);
+
+    assert_raft_log("#1: ", &nt.peers[&1].raft_log, (8, 0, 8));
+    assert_raft_log("#2: ", &nt.peers[&2].raft_log, (8, 0, 8));
+    assert_raft_log("#3: ", &nt.peers[&3].raft_log, (8, 0, 8));
+
+    // update to 8
+    assert_eq!(
+        (
+            nt.peers[&1].prs().get(1).unwrap().committed_index,
+            nt.peers[&1].prs().get(2).unwrap().committed_index,
+            nt.peers[&1].prs().get(3).unwrap().committed_index
+        ),
+        (0, 8, 8)
+    );
+
+    // #4 pass a smaller committed index, it occurs when the append response delay
+    let mut msg_append_response = new_message(2, 1, MessageType::MsgAppendResponse, 0);
+    msg_append_response.term = 3;
+    msg_append_response.index = 10;
+    msg_append_response.commit = 10;
+
+    nt.send(vec![msg_append_response.clone()]);
+
+    assert_eq!(nt.peers[&1].prs().get(2).unwrap().committed_index, 10);
```

Comment:
This seems a bug.

Reply:
shouldnt that commiitted index is update as 10?

Reply:
If leader doesn't have log 10, it shouldn't commit to 10. Giving that raft-rs doesn't take much effort in byzantine fault tolerant, I think remove the case is OK for now.